et_replay/comm/commsTraceParser.py

```diff
@@ -0,0 +1,205 @@
+        dtype = container_type.replace("GenericList[", "").replace("]", "")
+    else:
+        tensors.append(tensor_container[0])
+        dtype = container_type
+
+    msg_size = 0
+    for tensor in tensors:
+        msg_size += tensor[3]
+
+    return msg_size, dtype
+
+
+def _parseExecutionTrace(
+    in_trace: ExecutionTrace, target_rank: int, total_ranks: int
+) -> List:
+    """
+    Convert the Execution Trace comms metadata to the common trace format for replay.
+    """
+    ET_PG_NAME_TUPLE = in_trace.schema_pytorch() >= (1, 0, 3)
+    if (in_trace.schema_pytorch() < (1, 0, 3)):
+        raise ValueError(f"Only support trace version >1.0.3, but current trace version is {in_trace.schema.split('-')[0]}")
+
+    pg_ranks_map = _parse_proc_group_info(in_trace) # key is pg id, value is global ranks in this pg
+    comms_op_list = _parse_comms_op_node(in_trace, pg_ranks_map, target_rank, total_ranks)
+
+    return comms_op_list
+
+def _parse_proc_group_info(in_trace: ExecutionTrace):
+    pg_ranks_map = {} # {node_id : {process_group_id : [ranks] } }
+    for node in in_trace.nodes.values():
+        if "process_group:init" in node.name:
+            # info of this node is dumped using torch.distributed.distributed_c10d._world.pg_config_info
+            # at the start of profiling, but not not callback to torch.distributed.init_process_group()
+            # Pre-Assumption: all process groups has been created before profiling start.
+            try:
+                pg_objs = json.loads(node.inputs[0])
+            except json.decoder.JSONDecodeError:  # skip if pg_config_info is truncated
+                break
+            
+            pg_ranks_map[node.id] = {}
+            for pg in pg_objs:
+                if not pg["pg_name"].isdecimal():
+                    # TODO support local synchronization pg
+                    raise ValueError(f"Process group name is {pg['pg_name']} in node {node['id']}, which is not supported.")
+                (pg_id, ranks, group_size, group_count) = [pg[k] for k in ["pg_name", "ranks", "group_size", "group_count"]]
+                pg_id = int(pg_id)
+                pg_ranks_map[node.id][pg_id] = (
+                    ranks
+                    if len(ranks) > 0
+                    else list(range(group_size))
+                    # rank list is empty when all ranks are in a pg
+                )
+            break  # only one process_group init node per trace
+    return pg_ranks_map
+
+def _parse_comms_op_node(in_trace: ExecutionTrace, pg_ranks_map: dict, target_rank: int, total_ranks: int):
+    comms_op_list = []
+
+    for node_id in pg_ranks_map:
+        for pg_id, ranks in pg_ranks_map[node_id].items():
+            comm_args = _create_pg_init_node(node_id, pg_id, ranks, len(ranks))
+            comms_op_list.append(comm_args)
+
+    pg_ranks_map_flatten = {}
+    for k,v in pg_ranks_map.items():
+        pg_ranks_map_flatten.update(v)
+    
+    for node in in_trace.nodes.values():
```

Comment:
Please change this logic to directly use fields that are present inside the attributes of the node now.

If we see the older parsing logic uses fixed indices in the input. That is extremely flaky and not easy to maintain.
Earlier this year I added all these attributes as direct "args" in the node, if you see the json all the communication fields are now inside args besides inputs.

Read more here
https://github.com/pytorch/pytorch/issues/124674

First thing i would do is read these args here in execution_trace class
https://github.com/facebookresearch/param/blob/main/et_replay/lib/execution_trace.py#L108-L154

Then just use this commArgs structure here.

Reply:
Thanks, @briancoutinho ! Yes, it's better to use explicit fields of attributes. However, seems `seq` and `rank` are missing from attributes, still need to be parsed from inputs using fixed indices. Is this by design?
![image](https://github.com/user-attachments/assets/a1800f16-0a41-4a20-a57a-d2bf621779ea)
![image](https://github.com/user-attachments/assets/bd530fb9-31ba-4422-a86c-acdb7498a7e7)

Reply:
Done. Thanks!
Use the existing fields of attributes, but still parse `req` and `rank` from inputs, which don't exist in attributes now.